interfaces/smtp/v0/README.md

```diff
@@ -0,0 +1,64 @@
+# `smtp`
+
+## Overview
+
+This relation interface specification describes the expected behavior of charms integrating over the `smtp` Juju interface to provide or consume SMTP data.
+
+## Usage
+
+In most cases, this will be accomplished using the [smtp library](https://github.com/canonical/smtp-integrator-operator/blob/main/lib/charms/smtp_integrator/v0/smtp.py), although charm developers are free to provide alternative libraries as long as they fulfil the behavioural and schematic requirements described in this document.
+
+## Direction
+
+The `smtp` interface implements a provider/requirer pattern.
+The requirer is a charm that requires SMTP details to connect to an SMTP server, and the provider is a charm holding those details.
+
+```mermaid
+flowchart TD
+    Provider -- host, port, user, password_id, auth_type, transport_security, domain --> Requirer
```

Comment:
People have differing opinions about whether a username is secret or not. Is there a Canonical standard here? I think I have seen elsewhere that in a case like this the Juju secret would have all the credential, ie. both username and password, and the databag would only have the secret ID.

Reply:
Not that I'm aware of